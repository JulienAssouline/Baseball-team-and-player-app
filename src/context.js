import React, { useState, createContext, useEffect } from "react";
import axios from "axios";

export const TeamsContext = createContext();

export const TeamsProvider = props => {
  const [teamData, setTeamData] = useState(null);
  const [error, setError] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      try {
        const result = await axios(
          "https://statsapi.mlb.com/api/v1/teams?sportId=1"
        );
        setTeamData(result.data.teams);
      } catch (error) {
        setError(true);
      }
    };
    fetchData();
  }, []);

  return (
    <TeamsContext.Provider value={[teamData, error]}>
      {props.children}
    </TeamsContext.Provider>
  );
};
