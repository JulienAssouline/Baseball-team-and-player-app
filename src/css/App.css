.App {
  text-align: center;
  margin-left: 40px;
  margin-right: 40px;
}


body {
  background-color: #f5f5f5;
}

@media only screen and (max-width: 700px) {
  .App {
    margin-left: 0px;
    margin-right: 0px;
  }
}