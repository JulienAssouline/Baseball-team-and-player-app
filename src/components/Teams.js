import React, {useState, useEffect} from 'react';
import axios from "axios"
import '../css/teams.css'
import Paper from '@material-ui/core/Paper';

function Teams(props) {
    const [teamData, setTeamData] = useState([])

    useEffect(() => {
        const fetchData = async () => {
            const result = await axios("https://statsapi.mlb.com/api/v1/teams?sportId=1")
            setTeamData(result.data.teams)
        }

        fetchData()

    }, [])

    return (
        <div className="teams-container">
            {
                teamData.map((d,i) => (
                    <Paper onClick = {() => {
                        props.history.push({
                            pathname: `/team${d.id}`,
                            state: { team: d.name }
                        })
                    }} 
                    key = {d.id} 
                    className = "card-container">
                        <img className = "team-image" src = {`https://www.mlbstatic.com/team-logos/${d.id}.svg`} alt="Italian Trulli"/>
                        <h2> {d.name} </h2>
                    </Paper>
                ))
            }
        </div>
    );
}

export default Teams;
