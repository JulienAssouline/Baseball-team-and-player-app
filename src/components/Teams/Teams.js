import React, { useState, useEffect } from "react";
import axios from "axios";
import "../../css/teams.css";
import Paper from "@material-ui/core/Paper";

function Teams(props) {
  const [teamData, setTeamData] = useState(null);

  useEffect(() => {
    const fetchData = async () => {
      const result = await axios(
        "https://statsapi.mlb.com/api/v1/teams?sportId=1"
      );
      setTeamData(result.data.teams);
    };

    fetchData();
  }, []);

  if (!teamData) return <div data-testid="loading"> loading... </div>;

  return (
    <div className="teams-container">
      {teamData.map((d, i) => (
        <Paper
          onClick={() => {
            props.history.push({
              pathname: `/team${d.id}`,
              state: { team: d.name }
            });
          }}
          key={d.id}
          className="card-container"
        >
          <img
            className="team-image"
            src={`https://www.mlbstatic.com/team-logos/${d.id}.svg`}
            alt={`${d.name}`}
          />
          <h2 data-testid="team-name"> {d.name} </h2>
        </Paper>
      ))}
    </div>
  );
}

export default Teams;
