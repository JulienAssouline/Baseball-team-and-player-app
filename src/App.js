import React from 'react';
import './css/App.css'
import Teams from "./components/Teams"
import Roster from "./components/Roster"
import Player from "./components/Player"
import { BrowserRouter as Router, Route } from 'react-router-dom'



function App() {
  return (
      <div className="App">
        <Router>
          <Route path='/' exact component={Teams} />
          <Route path = "/team:id" exact component = {Roster}/>
          <Route path = "/player:id" exact component = {Player} /> 
        </Router>
      </div>

  );
}

export default App;
