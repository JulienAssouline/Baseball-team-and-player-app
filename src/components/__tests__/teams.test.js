import React from "react";
import { render, cleanup, waitForElement } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import axiosMock from "../../__mocks__/mock-axios";
import Teams from "../Teams/Teams";
import { TeamsContext } from "../../context";
import App from "../../App";

afterEach(cleanup);

test("loads and displays teams data", async () => {
  axiosMock.get.mockResolvedValueOnce({
    data: [{ id: 1, name: "Oakland Athletics" }]
  });

  const { getByTestId, getAllByTestId, queryByTestId } = render(
    <App>
      <TeamsContext.Provider>
        <Teams />
      </TeamsContext.Provider>
    </App>
  );

  expect(getByTestId("loading").textContent.trim()).toBe("loading...");

  expect(queryByTestId("error")).toBeNull();

  const teamName = await waitForElement(() => getAllByTestId("team-name"));
  expect(teamName.length).toBe(30);
  expect(teamName[0].textContent.trim()).toBe("Oakland Athletics");
});

test("renders images", async () => {
  const { getAllByTestId } = render(
    <App>
      <TeamsContext.Provider>
        <Teams />
      </TeamsContext.Provider>
    </App>
  );

  const teamImage = await waitForElement(() => getAllByTestId("image-test"));
  expect(teamImage.length).toBe(30);
  expect(teamImage[0].tagName.toLowerCase()).toBe("img");
});
