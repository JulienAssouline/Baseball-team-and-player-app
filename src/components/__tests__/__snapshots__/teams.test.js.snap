// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      data-testid="loading"
    >
       loading... 
    </div>
  </div>
</div>
`;
